src/graphdb.js

```diff
@@ -52,20 +52,50 @@ function toPascalCase (str) {
     return r.trim();
 }
 
+// Changes every instance of invalid characters in the given label with the following delimiters
+function replaceCleanseLabel(label) {
+    return label
+        .replaceAll("!", "_ex_")
+        .replaceAll("$", "_dol_")
+        .replaceAll("&", "_amp_")
+        .replaceAll("(", "_op_")
+        .replaceAll(")", "_cp_")
+        .replaceAll(".", "_dot_")
+        .replaceAll(":", "_cn_")
+        .replaceAll("=", "_eq_")
+        .replaceAll("@", "_at_")
+        .replaceAll("[", "_os_")
+        .replaceAll("]", "_cs_")
+        .replaceAll("{", "_oc_")
+        .replaceAll("|", "_vb_")
+        .replaceAll("}", "_cc_")
+        .replaceAll("-", "_hy_");
+}
+
+function checkInvalidChar(label) {
+    let identify = /[:.-]/;
```

Comment:
This function doesn't account for all of the special characters that are replaced. But instead of using this function you can just check if the final label is different from the original label.

Reply:
changed it to check if the final label is different from the original label but didn't add alias as a variable because some event loops require different aliases